### PR TITLE
Restrict access UI with OIDC

### PR DESCRIPTION
**Pull request by lurumad**
**What this PR does / why we need it**:

Add an example for how to restrict access for users on HealthChecks UI using an OpenId Connect server.

**Which issue(s) this PR fixes**:

Role Based Healthcheck API #646

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -581,6 +577,14 @@ HealthChecks can be used as [Release Gates for Azure DevOps](https://docs.micros
 
 Check this [README](./extensions/README.md) on how to configure it.
 
+## Protected HealthCheck.UI with OpendId Connect
```
Comment (CarlosLanderas) [inline README.md:580]: HealthChecks.UI

File: README.md [hunk 2]
```diff
@@ -581,6 +577,14 @@ HealthChecks can be used as [Release Gates for Azure DevOps](https://docs.micros
...
+
+There are some scenarios where you can find usefull to restrict access for users on HealthChecks UI, maybe for users who belong to some role, based on some claim value etc.
+
+We can leverage the ASP.NET Core Authentication/Authorization features for easily implement it. You can see a fully functional example using IdentityServer4 [here](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/tree/master/samples/HealthChecks.UI.Oidc) but you can use Azure AD, Auth0, Okta, etc.
```
Comment (CarlosLanderas) [inline README.md:584]: features **to** easily implement it

**Comment by CarlosLanderas** [comment]
Thanks for the PR @lurumad

File: samples/HealthChecks.UI.Oidc/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                })
+                .AddAuthentication(options =>
+                {
+                    options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                    options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+                })
+                .AddCookie(options =>
+                {
+                    options.SlidingExpiration = true;
+
```
Comment (sungam3r) [inline samples/HealthChecks.UI.Oidc/Startup.cs:39]:
```suggestion
```

**Pull request merged**